WcaOnRails/app/views/competitions/_competition_form.html.erb

```diff
@@ -205,6 +205,34 @@
         <%= f.input :currency_code, collection: Money::Currency.table.values, label_method: lambda { |c| "#{c[:name]} (#{c[:iso_code]})" }, value_method: lambda { |c| c[:iso_code] }, disabled: disable_money_input  %>
         <%= f.input :base_entry_fee_lowest_denomination, as: :money_amount, currency: @competition.currency_code, disabled: disable_money_input, currency_selector: "#competition_currency_code" %>
 
+        <script>
+          $('input[name="competition[base_entry_fee_lowest_denomination]"]').on('change', function() {
+            if(!$('#due_estimate').length){
+              $(this).siblings('.help-block').append('<br><p id="due_estimate"><b>XDXD</b></p>');
+            }
+
+            <% iso2 = Country.where(id: @competition.countryId).map(&:iso2) %>
+            <% countryBand = CountryBand.where(iso2: iso2).map(&:number)[0] %>
+            <% pricePerCompetitor = 0 %>
+
+            <% if !countryBand.nil? && countryBand > 0 %>
+              <% pricePerCompetitor = CountryBand::BANDS[countryBand][:value] %>
+
+              //Need a way to convert whatever the currency is to USD
```

Comment:
In line `205` above, the input field for `currency_code` is being created. You should be able to read the currency's iso_code from that field, and then use it to work with `Money::Currency` in Rails here.

But why do you want to convert into USD in the first place? For planning a competition budget, it would probably makes more sense to display the estimate in the local currency. You could optionally do something along the lines of:
`Estimated: 1234JPY (equivalent to 12.34USD as of %today)`.

Reply:
I think it's because the dedicated cost of each `countryBand` is in USD, the dues calculation would only be accurate if they're using the same currency. But yeah it would be convenient to convert it back as well to display in local money.